src/inference/enumerate.ad.js

```diff
@@ -17,8 +17,18 @@ module.exports = function(env) {
   function Enumerate(store, k, a, wpplFn, options) {
     util.throwUnlessOpts(options, 'Enumerate');
     options = util.mergeDefaults(options, {
-      maxExecutions: Infinity
+      maxExecutions: Infinity,
+      throwOnError: true,
+      maxRuntimeInMS: Infinity,
+      maxEnumTreeSize: Infinity
```

Comment:
`maxEnumTreeSize` is used at the very beginning of the enumeration, when there's no partial distributions available.